### ISSUE TITLE
Enhance Terra Clinical Result to Support Rich Text Values

### ISSUE THREAD

**Issue by jn050665** [cerner/terra-clinical#715]
# Issue Description
The Charting team is requesting the Terra Clinical Result to support rich text values. We use the Clinical Result in our detail panels, flowsheet, pop-ups, etc. All of the the result types we show are support except for rich text. We would like support for how this is displayed in the cell vs slide panel with full details. Today there is a gap in functionality because users cannot see the result as it was documented by the user with formatting.

**Comment by neilpfeiffer** [comment]
_Adding pre-work discussion comments and notes:_

\
**General:**
- Rich Text is a complex document formatting language that would require importing third party libraries that will parse and render rich text syntax as React content. Large investigation will need to be done in terms of what features in rich text need to be supported, how they would need to be themes, and how it is commonly used in clinical content across the consumer base for Terra. Since the project team requesting this is closest to the actual content needing to be managed, it might be best suited that a project team and not Terra take ownership of rich text implementation, which will give them first handle ability to fix, add, and adjust features as needed as they see issues within their project. Terra is happy to figure out how to best sort out which implementation details with the project team that gives the project team the most latitude, while doesn’t open up any supporting components to accidental misuse by another Terra consumer without as much firsthand knowledge. 

\
 **Possible Implementation Ideas:**
- 1) Terra adds built-in first-class support for rich text content and creates a new clinical-result sub-component specifically for this feature. The complexity for this would be high, and likelihood to be completed in the near term is fairly low, relative to other current requests.
- 2) Terra either opens up the single `ClinicalResult` component to be able to receive a node instead of text/number strings, or creates a new subcomponent that exclusively receives a node as its result content. The complexity for this would be low for Terra, and high for the consuming project team, though the comfortability of supporting this style of feature is also low for Terra.
- 3) Terra works with the internal team to help them make an internal project component that can be well controlled by the project team, Terra working with the team to expose any needed parts to keep the Terra and project components sharing similar requirements. The complexity for this would be low for Terra, high for the consuming project team. Terra would be the most comfortable supporting this style of implementation.

\
**Risks:**
- **_Theming_**: if Terra opens up the component to receive a node, theming (particularly the lowlight themes) will need to be handled by the consumer, or by making the block of content intentionally unthemed (not recommended).
- **_Node_**: if Terra opens up the component to receive a node, the ability enforce strict layouts is no longer possible. Consumers have the flexibility to include any content they want as a result, but further work would need to be done inorder to make sure the other supporting elements in the clinical-result component would work with any type of content provided to it.  

\
**Missing Requirements:**
  - **_Components needed_**: it will need to be clarified which of the clinical result components need to handle Rich Text. Is it just the base `ClinicalResult` single result component used for detail panels and on-page use, or would it also be needed in the `FlowsheetResultCell`, which will be extremely challenging to display in the size allowed.
- **_Construction/layout_**: designs need to be created and approved by the Clinical content leadership team on how the result should be constructed, so it is clear how it looks when also needing to show the additional icons for `Unverified`, `Modified` and `hasComment`, the additional displays below for DateTime or Result Concept, and if it is possible to be truncated.

**Comment by mjhenkes** [comment]
We met with the team requesting this change and are leaning towards option 2 or 3 as specified above.

We asked the team to conduct an audit of where they're using this component such that we can understand how best to restructure it but the general idea is:

* Create a new result that takes a node and arrange the clinical result icons accordingly (not inline).
  * We need to determine if all non flowsheet clinical results should have the re-located icons, if so we may not need a new component.
* Alternatively we could offer a component that is just the icons and leave the positioning up the the consuming team.

* This change should not affect the flowsheet cell.

This change is requested by mid 2021.
